docs/configuration.md

```diff
@@ -1249,6 +1251,18 @@ one of the `allow` regular expressions **and** one of the following holds:
 2.  `deny` is set but no URLs within the manifest match any of the `deny` regular
     expressions.
 
+### `indexes`
+
+Use the `indexes` subsection to configure validation of image indexes.
+If `disabled` is `false`, the validation allows nothing.
```

Comment:
What is `disabled`?

Reply:
`disabled` is the value of the `validation.disabled` flag in the config file. That flag seems to be designed to disable the whole validation section, but for this flag, disabling the section means leaving the default behaviour, which is to validate.

I guess really, if validation.disabled is set, then the registry shouldn't validate that manifests exist for the references in the index. That would change the existing behaviour though, where at the moment the registry _always_ validates that manifests exist, regardless of the `validation.disabled` flag.

Maybe it should be:
```
validation:
  disabled: true // If false this would allow sparse indexes
  indexes:
    referencesexist: true // true by default, but allows for people to turn it off
```

What do you think?

Reply:
I was thinking about this again.

1. If we go with this version, the config seems awkward:
```
validation:
  disabled: true // If this is true default behaviour wins, which is no sparse indexes. If its false the validation.indexes.sparse is used
  indexes:
    sparse: false // false by default
```

2. If we go with this version, the behaviour of the registry changes for anyone who had previously set validation.disabled=true, but the config feels more sane
```
validation:
  disabled: true // If true this disables manifest existence checks (enables sparse indexes). If false the validation.indexes.referencesexist is used
  indexes:
    referencesexist: true // true by default, but allows for people to turn it off to enable sparse indexes
```

3. If neither of those feel right, we could add a whole new block outside of validation:
```
indexes:
  sparse: false // false by default
```

4. We could put it in the storage block:
```
storage:
  sparseindexes: false // default false
```

I think my preference would be 2.

Reply:
@SteveLasker mentioned this would be good if it could validate for specific architectures.

For that, maybe config like this would work:
```yaml
validation:
  imageindexes: // Use image indexes because not all indexes are for multi-arch
    referencesexist: ['amd64'] // Architectures to validate existence of, can use [] for "validate none"
```